debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java

```diff
@@ -777,4 +780,16 @@ private static int validateMessageKeyColumnsField(Configuration config, Field fi
         }
         return problemCount;
     }
+
+    private static int validateServerName(Configuration config, Field field, Field.ValidationOutput problems) {
+        String serverName = config.getString(SERVER_NAME);
+
+        if (serverName != null) {
+            if (!SERVER_NAME_PATTERN.asPredicate().test(serverName)) {
+                problems.accept(SERVER_NAME, serverName, serverName + " has invalid format (expecting '" + SERVER_NAME_PATTERN.pattern() + "')");
```

Comment:
Let's make it a bit more user-friendly and say something like the following, instead of the regexp: ... has invalid format (only the underscore and alpha-numerical characters are allowed)